pkg/bpf/map_linux.go

```diff
@@ -124,54 +112,99 @@ type Map struct {
 	events *eventsBuffer
 }
 
-// NewMap creates a new Map instance - object representing a BPF map
-func NewMap(name string, mapType MapType, mapKey MapKey, keySize int,
-	mapValue MapValue, valueSize, maxEntries int, flags uint32, dumpParser DumpParser) *Map {
+func (m *Map) Type() ebpf.MapType {
+	if m.m != nil {
+		return m.m.Type()
+	}
+	if m.spec != nil {
+		return m.spec.Type
+	}
+	return ebpf.UnspecifiedMap
+}
+
+func (m *Map) KeySize() uint32 {
+	if m.m != nil {
+		return m.m.KeySize()
+	}
+	if m.spec != nil {
+		return m.spec.KeySize
+	}
+	return 0
+}
+
+func (m *Map) ValueSize() uint32 {
+	if m.m != nil {
+		return m.m.ValueSize()
+	}
+	if m.spec != nil {
+		return m.spec.ValueSize
+	}
+	return 0
+}
 
-	if size := reflect.TypeOf(mapKey).Elem().Size(); size != uintptr(keySize) {
-		panic(fmt.Sprintf("Invalid %s map key size (%d != %d)", name, size, keySize))
+func (m *Map) MaxEntries() uint32 {
+	if m.m != nil {
+		return m.m.MaxEntries()
+	}
+	if m.spec != nil {
+		return m.spec.MaxEntries
 	}
+	return 0
+}
 
-	if size := reflect.TypeOf(mapValue).Elem().Size(); size != uintptr(valueSize) {
-		panic(fmt.Sprintf("Invalid %s map value size (%d != %d)", name, size, valueSize))
+func (m *Map) Flags() uint32 {
+	if m.m != nil {
+		return m.m.Flags()
+	}
+	if m.spec != nil {
+		return m.spec.Flags
 	}
+	return 0
+}
+
+// NewMap creates a new Map instance - object representing a BPF map
+func NewMap(name string, mapType ebpf.MapType, mapKey MapKey, mapValue MapValue,
+	maxEntries int, flags uint32, dumpParser DumpParser) *Map {
+
+	keySize := reflect.TypeOf(mapKey).Elem().Size()
```

Comment:
You've flagged this in a previous PR as well. Currently, all code assumes that MapKey and MapValue are implemented on pointer receivers. Panicking isn't a friendly way of enforcing that, but this also isn't too high up on the list of things to tackle. We can make this friendlier when NewMap() is changed to take a MapSpec, but eventually, bpf.Map should no longer hold MapKey and MapValue, so this would be somewhat wasted effort.